### DISCUSSION TITLE
Swapping Libraries while maintaining internal tags

### DISCUSSION THREAD

**Discussion by hawkiboy** [quodlibet/quodlibet#3726] [Q&A] [answered]
I have been looking for a way to keep separate copies of my music library and be able to swap between them. Is there any way to tell QL that the path has changed but the library remains the same? 
My plan is to keep a few internal tags in QL, basically Rating and Comment (is there a way to add an internal tag?). I want to be able to update these tags and have them still work for me when I switch locations to an identical library with a different location.
I have tried creating an alias for the root of the library and using that in QL, but it doesn't seem to like that very well.

**Comment by declension** [comment, marked as answer]
No symlinks / aliases generally make the problem worse as QL knows too much about them (long story).

This sounds like https://github.com/quodlibet/quodlibet/issues/1939 ?
In which case, there's now a _Move Library_ button in the prefs in recent (4.4+) versions, you could try that perhaps

**Reply by hawkiboy** [reply]
OK, thanks. I really want an option that lets me store more info in the local QL database rather than in the file. I want to be able to make comments on tracks and not lose them every time I reload my library.

**Reply by declension** [reply]
OK - be aware all non-internal tags (including, say, `comment`) are stored in files anyway. Many users opt to have some of the more internal tags, e.g. `~#rating` and `~playcount` persisted into files too, but that's an option (see preferences), not least as it generates a lot more churn on files (e.g. for backups)

**Reply by hawkiboy** [reply]
I am surprised this is not a more requested feature (or maybe it is). What I really want I suppose is a database of 'music' regardless of whether or not I own it. Perhaps there is something like it in the musicbrainz world, or I may need to create it myself (I started with pyside2 but keep thinking there must be a solution out there). A want a way to tag the metadata so that even if I lose my files I could still have the metadata db (in the cloud) that would allow me to reconstruct. I suppose some of the streaming services probably have something like that, but I want to Also be able to associate it with my music library and use it to create playlists. Anyway, thanks for your help!
